### ISSUE TITLE
--config-yaml fails if config doesn't have yaml extension

### ISSUE THREAD

**Issue by amravyan** [vectordotdev/vector#10186]
Found and interesting issue with vector validate command: it fails if i specify `--config-yaml` but config actually doesn't have yml extension
Could you please investigate this bug?

v0.18.0
```
 [root@test2 ~]# vector validate --config-yaml /root/.ansible/tmp/ansible-tmp-1637918351.5709512-273358672672511/source
 Failed to load ["/root/.ansible/tmp/ansible-tmp-1637918351.5709512-273358672672511/source"]
 -------------------------------------------------------------------------------------------
 x expected an equals, found a colon at line 12 column 9
  
 [root@test2 ~]# vector validate --config-yaml /root/.ansible/tmp/ansible-tmp-1637918351.5709512-273358672672511/source.yml
 √ Loaded ["/root/.ansible/tmp/ansible-tmp-1637918351.5709512-273358672672511/source.yml"]
 √ Component configuration
 √ Health check "output-kafka"
 -----------------------------------------------------------------------------------------
                                                                                  Validated
 [root@test2 ~]# diff -s /root/.ansible/tmp/ansible-tmp-1637918351.5709512-273358672672511/source /root/.ansible/tmp/ansible-tmp-1637918351.5709512-273358672672511/source.yml
 Files /root/.ansible/tmp/ansible-tmp-1637918351.5709512-273358672672511/source and /root/.ansible/tmp/ansible-tmp-1637918351.5709512-273358672672511/source.yml are identical
```

**Comment by darren-fhf** [comment]
This appears to have been fixed by  e14acc7f30.

I am unable to reproduce the problem with 0f1aa9a6bd.

**Comment by spencergilbert** [comment]
Thanks @darren-fhf - this does appear to be resolved on the latest release. @amravyan please let us know if you're still running into this issue.
